File: output/ChatGPT/codenet/Python/Java/s975943078.java
import java.util.*;

public class s975943078 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int K = scanner.nextInt();
        int[] A = new int[N];
        for (int i = 0; i < N; i++) {
            A[i] = scanner.nextInt() % K - 1;
        }
        
        int[] sumA = new int[N + 1];
        int count = 0;
        Map<Integer, List<Integer>> modK = new HashMap<>();
        modK.put(0, new ArrayList<>());
        
        for (int i = 0; i < N; i++) {
            sumA[i + 1] = (sumA[i] + A[i]) % K;
            
            if (!modK.containsKey(sumA[i + 1])) {
                modK.put(sumA[i + 1], new ArrayList<>());
            }
            modK.get(sumA[i + 1]).add(i + 1);
        }
        
        for (int key : modK.keySet()) {
            List<Integer> values = modK.get(key);
            int lenK = values.size();
            
            for (int i = 0; i < lenK; i++) {
                int a = values.get(i);
                count += binarySearch(values, a + K, i, lenK) - i - 1;
            }
        }
        
        System.out.println(count);
    }
    
    private static int binarySearch(List<Integer> list, int target, int start, int end) {
        int left = start;
        int right = end;
        
        while (left < right) {
            int mid = (left + right) / 2;
            
            if (list.get(mid) < target) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        
        return left;
    }
}

File: output/ChatGPT/codenet/Python/C++/s975943078.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <map>

using namespace std;

int solve() {
    int N, K;
    cin >> N >> K;
    vector<int> A(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i];
        A[i] = (A[i] % K) - 1;
    }
    vector<int> sum_A(N + 1);
    vector<int> count(K);

    int ans = 0;
    map<int, vector<int>> modK;
    modK[0].push_back(0);
    for (int i = 0; i < N; i++) {
        sum_A[i + 1] = (sum_A[i] + A[i]) % K;
        modK[sum_A[i + 1]].push_back(i + 1);
    }

    for (auto it = modK.begin(); it != modK.end(); it++) {
        int key = it->first;
        vector<int>& values = it->second;
        int lenK = values.size();
        for (int i = 0; i < lenK; i++) {
            ans += upper_bound(values.begin(), values.end(), values[i] + K) - values.begin() - i - 1;
        }
    }

    cout << ans << endl;
    return 0;
}

int main() {
    solve();
}